functions.md

```diff
@@ -531,6 +531,30 @@ flatten([1, [2, [3, [4, []]]]]) => [1,2,3,4]
 flatten(null)                   => null
 ```
 
+### _all(array) -> boolean_
+
+True iff all elements of `array` evaluates to `true`
+
+Examples:
+
+```
+all([true, true, true])         => true
+all([true, true, false])        => false
+all(null)                       => false
```

Comment:
I think the last case should be `null`. In a test that counts as `false`, but it's good to maintain the distinction.

We also need to decide what `all([])` should evaluate to. I guess the real intention here is that we want it to be `false` only if there is a `false` value in the array. So I think the empty case should evaluate to `true`.

Reply:
I've been thinking about this and maybe the better way would be to take advantage of the truthy / falsy  and just return the last element if all elements were truthy or the first falsy element
```
all(["a", "b", 20]) => 20
all(["a", 0, 20]) => 0 , since 0 is falsy
all(["a", null, 20]) => null, since null is falsy
all(null) => null 
all([]) => true
```

Reply:
I tend to prefer actually returning a boolean, because that way we know the output type. This will have benefits when we compile to JVM bytecode. I know people sometimes write `x or y` and similar to get the first usable value that comes out, but we can do that with `fallback`. 

I'm a bit skeptical that this is something usable, since there's nothing inherently special about the first true or false value, as the input will usually be a dynamic array. To verify I checked the behaviour in Python, and `any([1,2,3,4]) => True`.

So I think the proposed return value is the right one.